src/test/java/BaseTest.java

```diff
@@ -1,12 +1,77 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
 
+import java.time.Duration;
 
-public class BaseTest {
 
+public class BaseTest {
+    static WebDriver driver;
+    static WebDriverWait wait;
+    static String url=null;
+    public Actions actions;
+    public String playlistName = "TEST2";
     @BeforeSuite
-    static void setupClass() {
+    public static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters({"baseURL"})
+    public void launchBrowser (String baseURL){
+        ChromeOptions ops = new ChromeOptions();
+        ops.addArguments("--remote-allow-origins=*");
+        driver= new ChromeDriver(ops);
+        actions=new Actions(driver);
+        wait=new WebDriverWait(driver, Duration.ofSeconds(4));
+        url=baseURL;
+        driver.get(url);
+        driver.manage().window().maximize();
+    }
+    public void provideEmail(String email){
+        WebElement emailField=wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+    public void providePassword(String password){
+        WebElement passwordField=wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='password']")));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+    public void clickSubmit(){
+        WebElement submitButton=wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='submit']")));
+        submitButton.click();
+    }
+    public void logIn(String email,String password){
+        provideEmail(email);
+        providePassword(password);
+        clickSubmit();
+    }
+    public void doubleClick(){
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#playlists > ul > li:nth-child(4)")));
+        WebElement playlistElement=driver.findElement(By.cssSelector("#playlists > ul > li:nth-child(4)"));
+        actions.doubleClick(playlistElement).perform();
+    }
+    public void enterPlaylistName(){
+        WebElement inputField1=wait.until(ExpectedConditions.elementToBeClickable
+                (By.cssSelector("input[name='name']")));
+        inputField1.sendKeys(Keys.chord(Keys.COMMAND,"A",Keys.BACK_SPACE));
+        inputField1.sendKeys(playlistName);
+        inputField1.sendKeys(Keys.ENTER);
+    }
+    public boolean doesPlaylistExist(){
+        WebElement playlistElement=wait.until(ExpectedConditions.visibilityOfElementLocated
+                (By.xpath("//a[text()='"+playlistName+"']")));
+        return playlistElement.isDisplayed();
```

Comment:
You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/224934521-0610d3fe-284a-45f2-aa67-58e39e5d1d0f.png)

HINT

![image](https://user-images.githubusercontent.com/104761482/224935416-17a43ad9-23d6-4c50-be78-59f0fbf419b1.png)

![image](https://user-images.githubusercontent.com/104761482/224935063-bb3c8ec8-a921-483d-9c92-6aba2e9fc724.png)

Reply:
<img width="1792" alt="Screenshot 2023-03-14 at 8 29 27 PM" src="https://user-images.githubusercontent.com/126124586/225172781-ee4f0233-529c-401f-8127-4460dd79f983.png">
<img width="1792" alt="Screenshot 2023-03-14 at 8 29 47 PM" src="https://user-images.githubusercontent.com/126124586/225172804-d066195e-ba53-449a-acde-bdde81430b6a.png">



Test was working normal for me Homework21